function exception_class int | vector int
